logger/logger.go

```diff
@@ -20,36 +20,60 @@ package logger
 import (
 	"os"
 	"path"
-	"strings"
 
 	"github.com/sirupsen/logrus"
 	"github.com/sonatype-nexus-community/nancy/types"
 )
 
-const DefaultLogFilename = "nancy.combined.log"
-const TestLogfilename = "nancy.test.log"
+const defaultLogFilename = "nancy.combined.log"
 
-// DefaultLogFile can be overridden to use a different file name for upstream consumers
-var DefaultLogFile = DefaultLogFilename
+var DefaultLogFile = defaultLogFilename
 
-// LogLady can be obtained from outside the package, the name is a reference to the brilliant
-// actress in Twin Peaks
-var LogLady = logrus.New()
+var logLady *logrus.Logger
 
-func init() {
-	err := doInit(os.Args)
+// GetLogger will either return the existing logger, or setup a new logger
+func GetLogger(loggerFilename string, level int) *logrus.Logger {
+	if logLady == nil {
+		logLevel := getLoggerLevelFromConfig(level)
+		err := setupLogger(loggerFilename, &logLevel)
+		if err != nil {
+			panic(err)
+		}
+	}
+	return logLady
+}
+
+// LogFileLocation will return the location on disk of the log file
+func LogFileLocation() (result string, err error) {
+	result, _ = os.UserHomeDir()
+	err = os.MkdirAll(path.Join(result, types.OssIndexDirName), os.ModePerm)
 	if err != nil {
-		panic(err)
+		return
 	}
+	result = path.Join(result, types.OssIndexDirName, DefaultLogFile)
+	return
 }
 
-func doInit(args []string) (err error) {
-	if useTestLogFile(args) {
-		DefaultLogFile = TestLogfilename
+func setupLogger(loggerFilename string, level *logrus.Level) (err error) {
+	logLady = logrus.New()
+
+	if loggerFilename != "" {
+		DefaultLogFile = loggerFilename
+	} else {
+		DefaultLogFile = defaultLogFilename
+	}
+
+	if level == nil {
+		logLady.Level = logrus.ErrorLevel
+	} else {
+		// Done because report caller adds 20-40% overhead per logrus docs, only set this when we want to debug
```

Comment:
meh on `20-40% overhead` unless its super super noticeable id argue that this cli tool the more information we can output for debugging the better. Not like this is some high performance app.

Reply:
My only response is "dunno". Gonna defer to @DarthHater

Reply:
Yeah I just kinda limited it when I first did it. If y'all want, remove the if around it, I'm fine with the hit for some extra info.

Reply:
I'm gonna punt on this for now and leave as is, given we probably need to come up with a logrus replacement anyway.